aws/rust-runtime/aws-config/src/lib.rs

```diff
@@ -439,7 +440,29 @@ mod loader {
             mut self,
             identity_cache: impl ResolveCachedIdentity + 'static,
         ) -> Self {
-            self.identity_cache = Some(identity_cache.into_shared());
+            self.identity_cache = TriStateOption::Set(identity_cache.into_shared());
+            self
+        }
+
+        /// Do not configure a default identity cache.
+        ///
+        /// Starting with [`BehaviorVersion::v2024_03_28`] a default cache is
+        /// set if not provided. This can be disabled which will result in each
+        /// client using their own cache.
+        ///
+        /// # Examples
+        ///
+        /// Turn off creating a default identity cache:
+        /// ```no_run
+        /// # async fn create_config() {
+        /// let config = aws_config::from_env()
+        ///     .no_identity_cache()
+        ///     .load()
+        ///     .await;
+        /// # }
+        /// ```
+        pub fn no_identity_cache(mut self) -> Self {
```

Comment:
Does this behave differently from `loader.identity_cache(IdentityCache::no_cache())`?

Reply:
Yes. `IdentityCahe::no_cache()` is still an explicit cache being set whereas `no_identity_cache()` leaves it as "explicitly unset" so as to not enable the default cache. Open to bikeshedding on naming.

Reply:
Discussed offline and decided to remove this API for now as it can always be added later and users can still control this by setting a different cache per/client.